_rules/object-has-acessible-name-8fc3b6.md

```diff
@@ -38,12 +41,20 @@ The `object` element is not rendered for presentational purposes. If the `object
 
 ## Accessibility Support
 
-Non-supported media formats make screen readers render the text content of the element instead of other attributes.
+There are no major accessibility support issues known for this rule.
 
 ## Background
 
 Testing that the [accessible name][] describes the purpose of the `object` element is not part of this rule and must be tested separately.
 
+Non-supported media formats make screen readers render the text content of the element instead of other attributes. 
+
+`Object` elements without an accessible name are ignored by assistive technologies unless they have an [explicit role][].
+
+When the object resource is not loaded the fallback content is rendered as shown in the Inapplicable Example: "This `object` element does not need an accessible name because it loads no image, audio, or video.".
+
+The [MIME type][] of the resource embedded in the `data` attribute impacts how the [accessible name][] of the `object` is computed. For example, `object` embedding [image MIME type][] may use their `alt` attribute to compute their [accessible name][], but `object` embedding [audio or video MIME types][] may not.
```

Comment:
This actually seems to be browser dependent.
When trying
```html
<object data="./test-assets/shared/w3c-logo.png" alt="W3C logo"></object>
<object data="./test-assets/moon-audio/moon-speech.mp3" alt="Moon speech"></object>

<object data="./test-assets/shared/w3c-logo.png" alt="W3C logo" type="image/png"></object>
<object data="./test-assets/moon-audio/moon-speech.mp3" alt="Moon speech" type="image/png"></object>

<object data="./test-assets/shared/w3c-logo.png" alt="W3C logo" type="audio/mpeg"></object>
<object data="./test-assets/moon-audio/moon-speech.mp3" alt="Moon speech" type="audio/mpeg"></object>
```
Chrome does not give any of the objects a name (and NVDA on top of it does not announce it); but Firefox makes the images a `graphic` with name "W3C logo" (and NVDA exposes it as "graphic W3C logo"), while not giving a name to the audios.

---
So, it seems that the `type` attribute actually isn't used at this level. 
However, the use of `alt` for images is browser dependent, so this should be an Accessibility Support note rather than a Background note.
I can't find a normative reference for the naming of `object` element, so I can't be sure which one is right 😕 Would you like to open an issue on HTML AAM and ask their opinion?

Reply:
Sure - and we will let this one sit for now? 

What is the best place to ask? https://github.com/w3c/html-aam/issues?

Reply:
Yes, I'm rather wait for confirmation of what is the "good" solution so we know what we are actually suppose to test 😄 

Their issues list is indeed the place to ask.

Reply:
https://github.com/w3c/html-aam/issues/431

Reply:
`alt` isn't supported on the object element.  Seems if this is going to be called out, it should be noted that it is not an official naming mechanism.  

Seems a potential issue to file against HTML to officially support the use of `alt` on the element.

Reply:
Thanks @scottaohara!

Reply:
@HelenBurge So, my take would be:
* Not use `alt` on `object` in any Passed or Inapplicable example.
* Remove this note from Background
* Add a note to Accessibility Support saying that some UA/AT combination use `alt` on `object` embedding MIME `image`, despite `alt` not being allowed on `object` + link to [HTML spec of attributes allowed on object](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-object-element:concept-element-attributes). And so element may pass/fail depending on the UA/AT combination. (so, more or less the same content as this note, but presented a bit differently)
* We could have a Failed example with an `<object alt="…">`.

I wasn't part of the TF discussion around that, so not sure exactly what the request to add this were. It may be worth taking it again to the TF for further discussion.